spec/spec_helper.rb

```diff
@@ -27,6 +28,7 @@
   config.before(:suite) do
     DatabaseCleaner.strategy = :transaction
     DatabaseCleaner.clean_with(:truncation)
+    TestAfterCommit.enabled = true
```

Comment:
Very nice! Now I get it. Thank you Gleb

On Saturday, March 26, 2016, Gleb Mazovetskiy <notifications@github.com>
wrote:

> In spec/spec_helper.rb
> <https://github.com/thredded/thredded/pull/161#discussion_r57511861>:
>
> > @@ -27,6 +28,7 @@
> >    config.before(:suite) do
> >      DatabaseCleaner.strategy = :transaction
> >      DatabaseCleaner.clean_with(:truncation)
> > +    TestAfterCommit.enabled = true
>
> I've changed post_common.rb to use after_commit instead of after_save in
> one of the hooks:
>
> after_commit :notify_at_users, on: [:create, :update]
>
> This is because notify_at_users uses the id, which is not available
> before commit on create, so the background job might have failed once in a
> while previously:
>
> def notify_at_users
>    AtNotifierJob.queue.send_at_notifications(self.class.name, id)end
>
> However, the tests are currently run in a transaction, so the after_commit
> hooks do not fire in tests. This gem makes after_commit work in tests.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/thredded/thredded/pull/161/files/203fa68768ee0b9cc46429c735bd7efd53fca1da#r57511861>
>